HookCatcher/management/commands/newPRupdate.py

```diff
@@ -0,0 +1,85 @@
+import json
+import os
+
+import requests
+import yaml  # for parsing unicode json to strings
+from django.conf import settings  # database dir
+from django.core.management.base import BaseCommand, CommandError
+from HookCatcher.models import State
+
+STATES_FOLDER = 'states'  # folder within git repo that organizes the list of states
+# header of the git GET request
+GIT_HEADER = {
+    'Authorization': 'token ' + settings.GIT_OAUTH,
+}
+
+
+# function for accessing state JSON files and saving data into models
+def saveStates(gitBranch, gitPRnumber, gitCommit, gitSourceType):
+    # get the directory location of the states folder with the JSON states
+    gitContentURLtemp = os.path.join(settings.GIT_REPO_API, 'contents')
+    statesDir = STATES_FOLDER + '?ref=' + gitBranch
+    # example url https://api.github.com/repos/MingDai/kolibri/contents/states?ref=test-pr
+    gitContentURL = os.path.join(gitContentURLtemp, statesDir)
+    reqStatesDir = requests.get(gitContentURL, headers=GIT_HEADER)
```

Comment:
Also, may want to consider putting just the username in settings rather than the whole API URL. The reset of the URL can be defined in application logic.